### PR TITLE
drivers: serial: stm32: renable IT RXNE as part of graceful disable

### PR DESCRIPTION
**Pull request by manoj153**
async_rx_disable does not re-enable RXNE interrupt, it was disabled
during enable

Signed-off-by: Manojkumar Subramaniam <manoj@electrolance.com>

### CODE REVIEW

**Comment by manoj153** [comment]
@erwango  @FRASTM 

for some info why I wanted to do this, I have an use case, where after using the async_rx_uart, I would like to enable my shell, 
but then found out my shell isn't working anymore due to the interrupt has been disabled prior thus this PR. If there's a better way to handle this use-case let me know guys :)

**Review by FRASTM — COMMENTED** [review]
This uart_stm32_async_rx_disable function is used under the CONFIG_UART_ASYNC_API 
would the RXNE irq be triggered in that config ?

**Comment by manoj153** [comment]
> This uart_stm32_async_rx_disable function is used under the CONFIG_UART_ASYNC_API
> would the RXNE irq be triggered in that config ?

https://github.com/zephyrproject-rtos/zephyr/blob/947039d54f330455a5d873586e58433c24f7a25e/drivers/serial/uart_stm32.c#L1086-L1089

you may look at this line, that RXNE IRQ disabled

File: drivers/serial/uart_stm32.c [hunk 1]
```diff
@@ -905,6 +905,10 @@ static int uart_stm32_async_rx_disable(const struct device *dev)
 	data->rx_next_buffer = NULL;
 	data->rx_next_buffer_len = 0;
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
```
Comment (erwango) [inline drivers/serial/uart_stm32.c:908]: Since `LL_USART_DisableIT_RXNE` is not done under this switch in `uart_stm32_async_rx_enable`, let's not use here here for coherency sake.
If a IRQ config flag clean up is required, let's do it in a dedicated PR.

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/serial/uart_stm32.c (hunk 1 above)
Comment (manoj153) [reply drivers/serial/uart_stm32.c:908]: - there's no switch `uart_stm32_async_rx_enable` defined inside the function. switch you referring to?
- `LL_USART_DisableIT_RXNE` was actually called under `uart_stm32_async_rx_enable` , take a look permalink from intact main code, line 1084
https://github.com/zephyrproject-rtos/zephyr/blob/698a28663cb364b0601eabd48c73a206fe1f715c/drivers/serial/uart_stm32.c#L1061-L1092

sorry @erwango I think I may not understand you, please explain if my understanding is wrong.

Comment (erwango) [reply drivers/serial/uart_stm32.c:908]: I think the code addition is correct.
My request is to remove the `#ifdef CONFIG_UART_INTERRUPT_DRIVEN` directive for coherency with the rest of the code.

**Review by erwango — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by dcpleung — APPROVED** [review]

**Comment by erwango** [comment]
@cfriedt This one is ready for merge (CI issue is Clang)

**Comment by cfriedt** [comment]
@erwango - will have a look at it shortly.

**Comment by cfriedt** [comment]
@galak, @nashif - it appears that the one failure for this PR is mainly with a west checkout. It is green everywhere else. OK to merge or should I re-run all tests?

**Comment by erwango** [comment]
@manoj153 Would you mind rebasing in order to try to get CI passed ?

**Comment by manoj153** [comment]
> @manoj153 Would you mind rebasing in order to try to get CI passed ?

Done
